hamilton_examples/titanic/data_loaders.py

```diff
@@ -0,0 +1,67 @@
+from typing import Generator, Tuple, Dict, Any
+
+import pandas as pd
+from hamilton.function_modifiers import load_from, value, extract_columns, tag, config, \
+    extract_fields
+
+
+@tag(feedback="minibatch_data_generator")
+def minibatch_data_generator(batch_size: int) -> Generator[Tuple[pd.DataFrame, int], None, None]:
```

Comment:
This whole thing is clever, but slightly confusing, and could easily be wrapped up in a single decorator. What we do is:
1. Create a generator that we keep cached (the state changes, but it keeps getting passed in). This outputs the chunk of data and the iteration #. 
2. Have a downstream caller that calls next -- this gets called every iteration.

I think we can bundle that all up into a decorator:

```python
@load_from.chunked_csv(path=value('...'), chunk_size=32)
def data(raw_data: pd.DataFrame) -> pd.DataFrame:
   return raw_data...
```